src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -117,18 +119,33 @@ public void searchButtonClicked() {
                           noFilesFound.set(true);
                           searchInProgress.set(true);
                           successfulSearch.set(false);
-
                       })
                       .onFinished(() -> searchInProgress.set(false))
                       .onSuccess(newRoot -> {
                           root.set(newRoot);
                           noFilesFound.set(false);
                           successfulSearch.set(true);
                       })
-                      .onFailure(dialogService::showErrorDialogAndWait)
+                      .onFailure(handleFailure())
                       .executeWith(taskExecutor);
     }
 
+    private Consumer<Exception> handleFailure() {
```

Comment:
Why not simply `void handleFailure(Exception exception)` and `onFailure(this::handleFailure)`?

Reply:
good catch!